src/org/zaproxy/zap/GuiBootstrap.java

```diff
@@ -345,46 +345,54 @@ private static void setDefaultViewLocale(Locale locale) {
      * {@link java.net.InetAddress InetAddress}) preventing some ZAP options from being correctly applied.
      */
     private void setupLookAndFeel() {
-        if (lookAndFeelSet) {
-            return;
+    	if (lookAndFeelSet) {
+    	   return;
+    	}
+    	lookAndFeelSet = true;    	
+    	if (setLookAndFeel(System.getProperty("swing.defaultlaf"))) {
+           return;
         }
-        lookAndFeelSet = true;
 
-        String lookAndFeelClassname = System.getProperty("swing.defaultlaf");
-        if (lookAndFeelClassname != null) {
-            try {
-                UIManager.setLookAndFeel(lookAndFeelClassname);
-                return;
-            } catch (final UnsupportedLookAndFeelException
-                     | ClassNotFoundException
-                     | ClassCastException
-                     | InstantiationException
-                     | IllegalAccessException e) {
-                logger.warn("Failed to set the specified look and feel: " + e.getMessage());
-            }
-        }
+    	OptionsParam options = Model.getSingleton().getOptionsParam();
 
-        try {
-            // Set the systems Look and Feel
-            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+        if (setLookAndFeel(getLookAndFeelClassname(options.getViewParam().getLookAndFeel()))) {
+           return;
+        }
 
-            if (Constant.isMacOsX()) {
-                OsXGui.setup();
-            } else {
-                // Set Nimbus LaF if available
-                for (final LookAndFeelInfo info : UIManager.getInstalledLookAndFeels()) {
-                    if ("Nimbus".equals(info.getName())) {
-                        UIManager.setLookAndFeel(info.getClassName());
-                        break;
-                    }
-                }
-            }
-        } catch (final UnsupportedLookAndFeelException
-                 | ClassNotFoundException
-                 | InstantiationException
-                 | IllegalAccessException e) {
-            logger.warn("Failed to set the \"default\" look and feel: " + e.getMessage());
+        if (Constant.isMacOsX()) {
+        	OsXGui.setup();
+        }else if (setLookAndFeel(getLookAndFeelClassname("Nimbus"))) {
+               return;
+        }else {
```

Comment:
This else statement is not required (we want to set the system L&F if not able to set the Nimbus and also to set the expected L&F for macOS).

Reply:
@thc202 Alright, I will just change it. I hope this is the last change I have to do :)

Reply:
I think so.